server/athenian/api/align/mutations/goals.py

```diff
@@ -32,20 +38,49 @@ async def resolve_create_goal(
 
 
 @mutation.field("removeGoal")
+@sentry_span
 async def resolve_remove_goal(
     _: Any,
     info: GraphQLResolveInfo,
     accountId: int,
-    id: int,
+    goalId: int,
 ) -> Dict[str, Any]:
     """Remove a Goal and referring TeamGoal-s."""
     async with info.context.sdb.connection() as sdb_conn:
         async with sdb_conn.transaction():
-            await delete_goal(accountId, id, sdb_conn)
+            await delete_goal(accountId, goalId, sdb_conn)
     remove_status = GoalRemoveStatus(success=True)
     return remove_status.to_dict()
 
 
+@mutation.field("updateGoal")
+@sentry_span
+async def resolve_update_goal(
+    _: Any,
+    info: GraphQLResolveInfo,
+    accountId: int,
+    input: dict,
+) -> dict:
+    """Update an existing Goal."""
+    update = _parse_update_goal_input(input, accountId)
+    goal_id = input["goalId"]
+    result = {"goal": {"id": goal_id}}
```

Comment:
Let's define the models for both the input and the output and use them here.

Reply:
defined model for output, declared fields for input